trade-finance-manager-ui/server/controllers/case/amendments/bankDecision.controller.js

```diff
@@ -0,0 +1,323 @@
+      bankDecisionDateMonth: errorsObject.amendmentBankDecisionDateMonth,
+      bankDecisionDateYear: errorsObject.amendmentBankDecisionDateYear,
+    });
+  }
+
+  try {
+    const payload = { bankDecision: { effectiveDate: amendmentBankRequestDate } };
+
+    const { status } = await api.updateAmendment(facilityId, amendmentId, payload);
+
+    if (status === 200) {
+      return res.redirect(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/banks-decision/check-answers`);
+    }
+
+    console.error('Unable to add the bank\'s decision efective date');
+    return res.redirect(`/case/${dealId}/underwriting`);
+  } catch (err) {
+    console.error('There was a problem adding the bank\'s decision effective date', { response: err?.response?.data });
+    return res.redirect(`/case/${dealId}/underwriting`);
+  }
+};
+
+/**
+ * @param {*} req
+ * @param {*} res
+ * gets check your answers page for bank decision
+ */
+const getAmendmentBankDecisionCheckAnswers = async (req, res) => {
+  const { amendmentId, facilityId } = req.params;
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  if (!amendment?.amendmentId || !amendment?.bankDecision?.decision) {
+    return res.redirect('/not-found');
+  }
+
+  const { user } = req.session;
+  const isEditable = userCanEditBankDecision(amendment, user) && amendment.status === AMENDMENT_STATUS.IN_PROGRESS;
+
+  // formats received and effective date from epoch
+  const receivedDateFormatted = amendment.bankDecision?.receivedDate ? format(fromUnixTime(amendment.bankDecision.receivedDate), 'dd MMM yyyy') : '';
+  const effectiveDateFormatted = amendment.bankDecision?.effectiveDate ? format(fromUnixTime(amendment.bankDecision.effectiveDate), 'dd MMM yyyy') : '';
+
+  return res.render('case/amendments/amendment-add-banks-decision-check-answers.njk', {
+    amendment,
+    isEditable,
+    user,
+    decision: amendment.bankDecision.decision,
+    receivedDate: receivedDateFormatted,
+    effectiveDate: effectiveDateFormatted,
+  });
+};
+
+/**
+ * @param {*} req
+ * @param {*} res
+ * posts bank decision and completes process by changing status to complete and redirects to underwriting page
+ */
+const postAmendmentBankDecisionCheckAnswers = async (req, res) => {
+  const { _id: dealId, amendmentId, facilityId } = req.params;
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  if (!amendment?.amendmentId || !amendment?.bankDecision?.decision) {
+    return res.redirect('/not-found');
+  }
+
+  try {
+    // updates amendment with status to completed and submitted flag as true on bank decision
+    const payload = { status: AMENDMENT_STATUS.COMPLETED, bankDecision: { submitted: true } };
+
+    const { status } = await api.updateAmendment(facilityId, amendmentId, payload);
+
+    if (status === 200) {
+      return res.redirect(`/case/${dealId}/underwriting`);
```

Comment:
both this redirect and the one on line 307 send you to the underwriting tab. if that's the case, then we can do that only once - rather than twice. so there is no need to check if the status is `200`

Reply:
done